core/src/test/java/edu/wpi/grip/core/metrics/TimerTest.java

```diff
@@ -0,0 +1,87 @@
+package edu.wpi.grip.core.metrics;
+
+import edu.wpi.grip.core.events.TimerEvent;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for {@link Timer}.
+ */
+public class TimerTest {
+
+  private EventBus eventBus;
+
+  @Before
+  public void setUp() {
+    eventBus = new EventBus();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testStartedTwice() {
+    Timer timer = new Timer(eventBus, this);
+    timer.started();
+    timer.started();
+    fail("Timer should have thrown an exception");
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testStoppedWhenNotStarted() {
+    Timer timer = new Timer(eventBus, this);
+    timer.stopped();
+    fail("Timer should have thrown an exception");
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testStoppedTwice() {
+    Timer timer = new Timer(eventBus, this);
+    timer.started();
+    timer.stopped();
+    timer.stopped();
+    fail("Timer should have thrown an exception");
+  }
+
+  @Test
+  public void testTiming() throws InterruptedException {
+    Timer timer = new Timer(eventBus, this);
+    timer.started();
+    Thread.sleep(1000);
```

Comment:
Don't do this. This is why the `Stopwatch` API has a constructor that takes a `Ticker`. Explicitly so you don't have to do this.